### PR TITLE
fix(dolphinscheduler-alert): fix create http request error

### PR DESCRIPTION
**Pull request by hunter-cloud09**
This pull request is code cleanup without any test coverage.
request type should use equalsIgnoreCase to equal,msg should add URLEncoder.encode #14777 #14781

### CODE REVIEW

**Review by Radeity — APPROVED** [review]

**Comment by fuchanghai** [comment]
@Radeity plz update and rerun the failed part

**Review by fuchanghai — APPROVED** [review]
LGTM ,if pass CI

File: dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java [hunk 1]
```diff
@@ -139,7 +141,11 @@ private void setMsgInUrl(String msg) {
             if (!url.contains(URL_SPLICE_CHAR)) {
                 type = URL_SPLICE_CHAR;
             }
-            url = String.format("%s%s%s=%s", url, type, contentField, msg);
+            try {
+                url = String.format("%s%s%s=%s", url, type, contentField, URLEncoder.encode(msg, DEFAULT_CHARSET));
```
Comment (hunter-cloud09) [inline dolphinscheduler-alert/dolphinscheduler-alert-plugins/dolphinscheduler-alert-http/src/main/java/org/apache/dolphinscheduler/plugin/alert/http/HttpSender.java:145]: Hi, @Radeity @fuchanghai , If add this, Should I remove the 124 lines of code to construct the URL, and then directly construct the URI through url?

**Review by brave-lee — APPROVED** [review]
+1

**Pull request merged**